pymongo/collection.py

```diff
@@ -3030,6 +3030,9 @@ def find_one_and_update(self, filter, update,
           ...     return_document=ReturnDocument.AFTER)
           {u'_id': u'userid', u'seq': 1}
 
+        Regardless the value of **return_document**, if no document matches
+        the filter, the return value is ``None``.
```

Comment:
This is not true. For example when `upsert=True` and no document matched the filter a new document will be inserted and the return value will be `None` with `BEFORE` but it will be equal to the inserted document with `AFTER`.

This is why the documentation for `AFTER` down below says:        
"If :attr:`ReturnDocument.AFTER`, returns the updated or **inserted** document."

Do you think it would be an improvement to say:
"If :attr:`ReturnDocument.AFTER`, returns the updated or **upserted** document."

Reply:
Hi Shane. Good call.

The condition I was trying to document was the condition where `upsert=False` but `return_document` is `AFTER`, where the return value is `None`. I'll take another cut and try to address all these conditions together.